orchestrator/services/database_introspection.py

```
@@ -90,19 +96,37 @@ def introspect(self, include_samples: bool = True, sample_limit: int = 5) -> Dic
-
+                                logger.warning(
+                                    f"Failed to sample column {col_name} in {table}",
+                                    exc_info=True
+                                )
+                                try:
+                                    conn.rollback()
+                                except Exception:
+                                    logger.debug("Rollback failed or unnecessary", exc_info=True)
                     # Row count
                     try:
                         cnt = conn.execute(text(f"SELECT COUNT(*) FROM {schema}.\"{table}\""))
```

Ruff (0.14.5)

Possible SQL injection vector through string-based query construction

(S608)